src/Hl7.Fhir.Patch/Adapters/IObjectAdapter.cs

```diff
@@ -0,0 +1,122 @@
+/* 
+ * Copyright (c) 2020, Firely (info@fire.ly) and contributors
+ * See the file CONTRIBUTORS for details.
+ * 
+ * This file is licensed under the BSD 3-Clause license
+ * available at https://github.com/FirelyTeam/fhir-net-api/blob/master/LICENSE
+ */
+
+using Hl7.Fhir.Patch.Operations;
+
+namespace Hl7.Fhir.Patch.Adapters
+{
+    /// <summary>
+    /// Defines the operations that can be performed on a patch document.
+    /// </summary>
+    public interface IObjectAdapter
```

Comment:
I find the distinction between (I)ObjectAdapter and IAdapter confusing.

Reply:
Both interfaces implement Adapter pattern and add methods to perform each of the Patch Operations to `ElementNode` and `IEnumerable<ITypedElement>`. 

`ObjectAdapter` is an abstraction over the concrete implemetations for each of the types supported by Patch Operatons (`ElementNodeAdapter` and `CollectionAdapter` respectively). 

`IObjectAdapter` and `IAdapter` are esseantially the same interface, but they needed to be different to avoid duplicating code. Some of the patch operations is just a combination of others (e.g. Move == Delete + Insert). `IObjectAdapter` is a public interface, which defines methods for each of the patch operations, while `IAdapter` is an internal interface that only defines the base methods that are necessary to implement all the other patch operations.